Improve install-standalone and install-profile scripts

Clumsy-Coder opened the pull request:
- install-standalone
  - [x] check if config file exist
  - [x] print a banner line before running config
  - [x] load shell ENV from a file (currently not being used)
    - The code is implemented, it not enabled
  - [x] Add documentation on how the script works
- install-profile
  - [x] simplify code
    - remove redundant code of `install-standalone`
    - use a single command to run profile configs
  - [x] check if profile exists
  - [x] have the ability to run multiple profiles in a single command
  - [x] print line banner before running profile
  - [x] ignore certain lines in a profile
    - Allows the ability to omit a config from running in a profile
  - [x] Add documentation on how the script works
  - [x] Remove unused code
    - [x] Remove unused ENVs
- miscellaneous
  - [x] add example config
  - [x] add example profile
- Update README
  - [x] running a profile
    - [x] general info for running a profile
    - [x] running single profile
    - [x] running multiple profiles
    - [x] ignore certain lines in a profile
  - [x] running a config
    - [x] for general info for running a config
    - [x] for running single config
    - [x] for running multiple configs
    - [x] for running config in sudo
  - [x] using ENVs
    - [x] general info on using ENVs
    - [x] why use ENVs
    - [x] how to enable
  - [x] update file tree

Clumsy-Coder committed:
feat(install-standalone): check if config file exists

  ## what
  - Check if dotbot config file exists before running file

  ## how
  - loop through command line arguments
    - if file doesn't exist
      - print warning
      - flag to exit
  - check if exit flag is set
    - exit in non zero code
  - clear variables used

  ## why
  - Better to check beforehand if a config file exists.
  - Causes a bug if running dotbot config that doesn't exist

  ## where
  - ./install-standalone

  ## usage
  ./install-standalone <config ... >
  Ex: ./install-standalone astronvim

  ./install-standalone <config>-sudo
  Ex: ./install-standalone apt-sudo

Clumsy-Coder committed:
chore(install-standalone): print a line before running config

  ## what
  - print a line the length of terminal width before running the config

  ## how
  - use 'stty' get terminal width
  - use printf to print a character 'N' number of times

  ## why
  - Easier to disinguish between console output of one config to another

  ## where
  - ./install-standalone

  ## usage

Clumsy-Coder committed:
feat(install-standalone): load shell ENVs before running config

  ## what
  - load ENVs before running config
    - currently not in use

  ## how
  - source the file that loads ENVs
  - when running config in sudo
    - Pass env "HOME" to the current users HOME directory
    - Pass env to be loaded

  ## why
  - An ENV can be used in a config
    - Ex: setting ALACRITTY_CONFIG_PATH as an ENV, and then using it to
      link file
      Ex: in a YAML file
      ```yaml
      - link:
          $ALACRITTY_CONFIG_PATH: tools/alacritty/alacritty.yml
      ```
  - keep track of ENVs in one place when using them to create links,
    folders or being used in shell commands

  ## where
  - ./install-standalone

  ## usage

Clumsy-Coder committed:
docs(install-standalone): add comments on how the script works

Clumsy-Coder committed:
feat(install-profile): check if profile exists

  ## what
  - Check if dotbot profile file exists before continuing

  ## how
  - loop through command line arguments
    - if profile doesn't exist
      - print warning
      - flag to exit
  - check if exit flag is set
    - exit in non zero code
  - clear variables used

  ## why
  - Better to check beforehand if a profile file exists.

  ## where
  - ./install-profile

  ## usage
  ./install-profile < profile >
  Ex: ./install-profile profile1

Clumsy-Coder committed:
refactor(install-profile): simplify ./install-profile script

  ## what
  - simplify script

  ## how
  - remove identical code from ./install-standalone
  - run ./install-standalone to run all the configs from profile
    - use one command to run the configs rather than looping the configs
      and the running them

  ## why
  - ./install-profile and ./install-standalone have almost identical
    script.
  - better to reuse working code from ./install-standalone

  ## where
  - ./install-profile

  ## usage
  ./install-profile <profile>

Clumsy-Coder committed:
feat(dotbot): run multiple dotbot profiles in a single command

  ## what
  - Have the ability to run multiple profiles in a single command

  ## how
  - Loop through each profile argument
    - Extract configs of that profile
    - run dotbot configs using ./install-standalone

  ## why
  - Allows you to be able to run multiple dotbot profiles in a single command

  ## where
  ./install-profile

  ## usage
  ./install-profile <profile ... >

  Ex:
  ./install-profile profile1 profile2

Clumsy-Coder committed:
chore(install-profile): print line banner when running dotbot profile

Clumsy-Coder committed:
feat(dotbot): ignore commented lines in a profiles

  ## what
  - Have the ability to ignore commented lines in dotbot profiles

  ## how
  - Use grep to filter out commented lines
    - uses '#' as comment string to filter out
    - check https://stackoverflow.com/a/8197412/3053548

  ## why
  - Easier to ignore some of the dotbot configs in a profile without
    removing it from the file

  ## where
  - ./install-profile

  ## usage

Clumsy-Coder committed:
docs(install-profile): add comments on how the script works

Clumsy-Coder committed:
chore(install-profile): remove unused ENVs

Clumsy-Coder committed:
chore(profiles): add an example profile

  ## what
  - add an example profile

  ## how

  ## why
  - to showcase ignoring certain lines in a profile

  ## where

  ## usage

Clumsy-Coder committed:
chore(config): add an example config

  ## what
  - add an example profile

  ## how

  ## why
  - to showcase using ENVs

  ## where

  ## usage

Clumsy-Coder committed:
docs(readme): update readme documentation

Clumsy-Coder commented:
Pull request connected to #7

ecarlson94 commented:
I'm sorry it has taken me a while to get to this. I am in the process of moving. I will try to review this PR sometime this week.

kiriwalawren commented on README.md:
```diff
@@ -8,18 +9,30 @@ The structure of this repository is heavily inspired by [vsund](https://github.c
 
-- [Dependencies](#dependencies)
-- [Installation](#installation)
+- [Dotbot Template](#dotbot-template)
+  - [Table of Contents](#table-of-contents)
```
We don't need these.
```suggestion
```

kiriwalawren replied:
The rest will need to be de-dented after the removal of the above.

Clumsy-Coder replied:
Made changes as per requested.

Check [fb88d4f](https://github.com/ecarlson94/dotbot-template/pull/9/commits/fb88d4f5033a76023ff3644ad71c2b4bf7a6eafc)

kiriwalawren reviewed (changes requested):
Good job. I appreciate the documentation. Only thing I can see is that there are 2 unnecessary additions to the table of contents.

How have you tested these changes?

Clumsy-Coder committed:
docs(readme): remove top level heading table of contents

  ## what
  - remove heading from table of contents
    - 'Dotbot template'
    - 'Table of Contents'
  - De-dent (shift tab to the left) the rest of the table of contents

  ## how

  ## why
  - suggestions made by repo author (ecarlson94)
    - check https://github.com/ecarlson94/dotbot-template/pull/9#pullrequestreview-1058904033

  ## where

  ## usage
